crypto/x509/x_all.c

```diff
@@ -65,12 +65,14 @@ int X509_http_nbio(OCSP_REQ_CTX *rctx, X509 **pcert)
 
 int X509_REQ_sign(X509_REQ *x, EVP_PKEY *pkey, const EVP_MD *md)
 {
+    x->req_info.enc.modified = 1;
```

Comment:
> By the way, should `X509_REQ_sign{,_ctx}` also flag modified if `ASN1_item_sign{_ctx}` call was successful? 
> Then `X509_sign{,_ctx}` also to be changed I think. Currently I edited same as `X509_sign{,_ctx}`.

Indeed - please change all six functions such that they set the flag only on success.